src/main/java/com/vozniuk/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+import com.vozniuk.bookstore.repository.OrderRepository;
+import com.vozniuk.bookstore.repository.ShoppingCartRepository;
+import com.vozniuk.bookstore.repository.UserRepository;
+import com.vozniuk.bookstore.security.AuthenticationService;
+import com.vozniuk.bookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final AuthenticationService authenticationService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public OrderResponseDto save(OrderRequestDto requestDto) {
+        Order order = new Order();
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderDate(LocalDateTime.now());
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findByUserId(authenticationService.getUserId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart to place "
+                        + "an order by user with id: " + authenticationService.getUserId()));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        Set<OrderItem> orderItems = new HashSet<>();
+        BigDecimal total = BigDecimal.valueOf(0);
+        for (CartItem item : cartItems) {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+            orderItem.setBook(item.getBook());
+            orderItem.setPrice(item.getBook().getPrice());
+            orderItem.setQuantity(item.getQuantity());
+            orderItems.add(orderItem);
+            total = total.add(item.getBook().getPrice());
+        }
+        order.setUser(userRepository.findById(authenticationService.getUserId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find user with id: "
+                + authenticationService.getUserId())));
+        order.setOrderItems(orderItems);
+        order.setTotal(total);
+        order.setStatus(Order.Status.NEW);
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderResponseDto> findAll(Pageable pageable) {
+        return orderRepository.findAll(pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(Long id, OrderUpdateRequestDto requestDto) {
```

Comment:
In your current implementation, it is possible to access data of another user. For example, I`m user A, and in orderUpdate Status request I can specify the id of another user order. To avoid such situations you should check whether provided order id belongs to user it possible to implement it on java level by adding an if statement or on DB level by adding an additional Where clause, I prefer the second way. Also, please take into account that this check should be applied to all methods where described situation can happen

Reply:
Order status could be updated only by users with ADMIN role (regular buyers are not able to change status even of their own orders), so that's not an issue here. There is @PreAuthorize("hasRole('ROLE_ADMIN')") annotation in the corresponding controller method